A27-xds-global-load-balancing.md

```diff
@@ -0,0 +1,432 @@
+LB policy, described below.  The configuration of the CDS policy will
+indicate which `Cluster` resource will be used.
+
+Note that the resolver will return an empty list of addresses, because
+in the xDS API flow, the addresses are not returned until the
+`ClusterLoadAssignment` resource is obtained later.
+
+The xds resolver will also return a reference to the XdsClient object
+that it instantiated.  This reference will be passed down to the LB
+policies, which is how the resolver and LB policies can share the same
+XdsClient object.
+
+#### CDS LB Policy
+
+The CDS LB policy will be the top-level LB policy.  Its configuration
+will be of the following form:
+
+```
+{
+  "cluster": "<cluster name>"
+}
+```
+
+The CDS policy will use the XdsClient object passed in from the xds resolver
+to query for the `Cluster` resource with the cluster name from the LB
+policy configuration.
+
+Initially, the CDS policy will always create an EDS policy as the child
+policy.  In the future, this may change; for example, we may add support
+for clusters that use DNS or for aggregate clusters, which would require
+different child policies.
+
+#### EDS LB Policy
+
+The EDS LB policy will be the child of the CDS LB policy.  Its
+configuration will be of the following form:
+
+```
+{
+  "edsServiceName": "<EDS service name>",
+  // Optional; if not specified, load reporting will be disabled.
+  // If value is the empty string, that means to use the xDS server as
+  // the LRS server.  (Currently, we do not support non-empty values;
+  // support for this will be added later.)
+  "lrsLoadReportingServerName": "<LRS server name>"
+}
+```
+
+The EDS policy will use the XdsClient object passed in from the xds
+resolver to query for the `ClusterLoadAssignment` resource with the
+EDS service name from the LB policy configuration.
+
+Note that unlike the old grpclb policy, which just received a flat list
+of backend servers from the balancer to round-robin across, the EDS policy 
+receives a hierarchical list, where the servers are grouped into
+localities, and the localities are grouped into priorities.  The EDS
+policy will be responsible for both selecting the highest priority set
+of localities that are reachable and distributing the traffic across the
+localities in the selected priority based on the locality weights.
+
+Note that the EDS policy will *not* support
+[overprovisioning](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/overprovisioning),
+which is different from Envoy.  Envoy takes the
+overprovisioning into account in both [locality-weighted load
+balancing](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/locality_weight)
+and [priority
+fail-over](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/priority),
+but gRPC assumes that the xDS server will update it to redirect traffic
+when this kind of graceful fail-over is needed.
```

Comment:
Well, Envoy could also do this today, but it doesn't because it does its own health checking and uses that to compute the effective available capacity for each priority/level. Does gRPC do its own HC? If so, shouldn't it be doing that?